Modules/posixmodule.c

```diff
@@ -10060,6 +10065,252 @@ os_times_impl(PyObject *module)
+                                                          curr_value->it_value.tv_nsec);
+    PyObject *next_expiration = PyFloat_FromDouble(_next_expiration);
+    if (next_expiration == NULL) {
+        return NULL;
+    }
+    double _interval = CONVERT_SEC_AND_NSEC_TO_DOUBLE(curr_value->it_interval.tv_sec,
+                                                   curr_value->it_interval.tv_nsec);
+    PyObject *interval = PyFloat_FromDouble(_interval);
+    if (interval == NULL) {
+        Py_DECREF(next_expiration);
+        return NULL;
+    }
+    PyObject *tuple = PyTuple_Pack(2, next_expiration, interval);
+    Py_DECREF(interval);
+    Py_DECREF(next_expiration);
+    return tuple;
+}
+
+static PyObject *
+build_itimerspec_ns(const struct itimerspec* curr_value)
+{
+    long long _next_expiration = (long long)curr_value->it_value.tv_sec * ONE_SECOND_IN_NS +
+                    (long long)curr_value->it_value.tv_nsec;
+    PyObject *next_expiration = PyLong_FromLongLong(_next_expiration);
+    if (next_expiration == NULL) {
+        return NULL;
+    }
+    long long _interval = (long long)curr_value->it_interval.tv_sec * ONE_SECOND_IN_NS +
+                       (long long)curr_value->it_interval.tv_nsec;
+    PyObject *interval = PyLong_FromLongLong(_interval);
+    if (interval == NULL) {
+        Py_DECREF(next_expiration);
+        return NULL;
+    }
+    PyObject *tuple = PyTuple_Pack(2, next_expiration, interval);
+    Py_DECREF(interval);
+    Py_DECREF(next_expiration);
+    return tuple;
+}
+
+/*[clinic input]
+os.timerfd_create
+
+    clockid: int
+        A valid clock ID constant as timer file descriptor.
+
+        time.CLOCK_REALTIME
+            A settable system-wide real-time clock.
+            If system clock is changed, timer setting need to be updated.
+            See os.TFD_TIMER_ABSTIME and os.TFD_TIMER_CANCEL_ON_SET.
+
+        time.CLOCK_MONOTONIC
+           A nonsettable monotonically increasing clock.
+           Even if system clock is changed, timer setting will be not affected.
+
+        time.CLOCK_BOOTTIME
+           Same as time.CLOCK_MONOTONIC except that it inclues any time that
+           the system is suspended.
+    /
+    *
+    flags: int = 0
+        0 or a bit mask of os.TFD_NONBLOCK or os.TFD_CLOEXEC.
+
+        os.TFD_NONBLOCK
+            If *TFD_NONBLOCK* is set as a flag, read doesn't blocks.
+            If *TFD_NONBLOCK* is not set as a flag, read block until the timer fires.
+
+        os.TFD_CLOEXEC
+            If *TFD_CLOEXEC* is set as a flag, enable the close-on-exec flag
+
+Create and return a timer file descriptor.
+[clinic start generated code]*/
+
+static PyObject *
+os_timerfd_create_impl(PyObject *module, int clockid, int flags)
+/*[clinic end generated code: output=1caae80fb168004a input=f23bfa49fc4b6576]*/
+
+{
+    int fd;
+    Py_BEGIN_ALLOW_THREADS
+    fd = timerfd_create(clockid, flags);
```

Comment:
[PEP 446](https://peps.python.org/pep-0446/) requires that all FDs created by Python are non-inheritable. Please always pass TFD_CLOEXEC:
```suggestion
    flags |= TFD_CLOEXEC;  // PEP 446: always create non-inheritable FD
    fd = timerfd_create(clockid, flags);
```

Reply:
@vstinner 
Applied your suggestion.